### DISCUSSION TITLE
First version

### DISCUSSION THREAD

**Discussion by AlUlkesh** [AlUlkesh/sd_delete_button#1] [Announcements]
This is a quick first version. It currently works mostly for single generations. If multiple images are generated, the delete button deletes the last generated image (excluding grids).

Obviously that should be changed. But what would be the best logic?

**Comment by Aamir3d** [comment]
Going to install this and update with feedback.

**Comment by Aamir3d** [comment]
This works great - interrupting and image and deleting works.
Feedback:
1. Make the image deleted part message bold or highlighted (there was no way to know if the deletion was actioned till I saw the small text at the bottom of the image)
2. Another way to action deletions on-the-fly would be this suggestion https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/8456
3. For multiple images, I know the images are displayed as tiny thumbnails, maybe a CTRL-Select multiple and trash option would work? 
(This deletion methodology could work in the batch move/delete idea for image browser as well)

Thank you for doing this - it's a more productive to utilize the UI in this way if you're trying out so many new things with AI generated imagery.
Appreciate everything you do.

**Comment by Aamir3d** [comment]
https://www.reddit.com/r/StableDiffusion/comments/11pi11t/delete_trash_button_extension_for_automatic1111_ui/ 
If you're on Reddit, happy to mention you there.

**Comment by Aamir3d** [comment]
@AlUlkesh Quick piece of feedback
Would it be possible to change the position of the X button to under the Generate button?

It's not a deal breaker, but I can see how things slow down if you click "Interrupt", scroll slightly to hit Delete, and then go back to Generate

I'm using Delete regularly, but need to do a bit of mouse hopping to get from Generate to Delete and back again.
This would be also be related to the 8456 suggestion above.
